File: 0050-powx-n/0050-powx-n.py
class Solution:
    def myPow(self, x: float, n: int) -> float:
#         isNegative = False
#         if n < 0 : 
#             isNegative = True
#         ans = 1.0
#         n = abs(n)
#         for i in range(0,n) :
#             ans *= x
        
#         if isNegative :
#             return 1 / ans
        
#         return ans


            ans = 1.0
            flg = n < 0
            n = abs(n)
            while n > 0 :
                if n%2 == 1 :
                    ans *= x
                x *= x
                n //= 2
                
            return 1 / ans if flg else ans

File: 0050-powx-n/0050-powx-n.java
class Solution {
    public double myPow(double x, int n) {      
        
        if(n == 0) return 1.0;
        double res = 1.0;
        long power = n;
        
        if(power < 0){
            x = 1 / x;
            power = -1*power;
        }
        
        while (power > 0) {
            if (power % 2 == 1) {
                res *= x;
            }
            x *= x;
            power /= 2;
        }
        
        return res;
    }
}